Swiften/Elements/ClientState.h

```diff
@@ -18,18 +18,14 @@ namespace Swift {
 
         public:
             enum ClientStateType {Active, Inactive};
```

Comment:
Sorry for not noticing earlier. But could you change this to an enum class, also known as [scoped enum](http://en.cppreference.com/w/cpp/language/enum). This is basically a type safe version of the classic enums. 
Other than that, the PR looks fine and can be squashed together. Afterwards I'll merge it.